app/src/main/java/com/hedvig/app/util/featureflags/RemoteFeatureFlagProvider.kt

```diff
@@ -0,0 +1,46 @@
+package com.hedvig.app.util.featureflags
+
+import com.hedvig.app.feature.settings.Market
+import com.hedvig.app.feature.settings.MarketManager
+import com.hedvig.app.service.RemoteConfig
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+class RemoteFeatureFlagProvider(
+    private val marketManager: MarketManager,
+    private val remoteConfig: RemoteConfig,
+) : FeatureFlagProvider {
+
+    override val priority = REMOTE_PRIORITY
+
+    private var seCampaignVisible: Boolean = false
+    private var noCampaignVisible: Boolean = false
+    private var dkCampaignVisible: Boolean = false
+
+    init {
+        CoroutineScope(Dispatchers.IO).launch {
+            seCampaignVisible = remoteConfig.fetch().seCampaignVisible
+            noCampaignVisible = remoteConfig.fetch().noCampaignVisible
+            dkCampaignVisible = remoteConfig.fetch().dkCampaignVisible
+        }
+    }
+
+    override fun isFeatureEnabled(feature: Feature) = when (feature) {
```

Comment:
Since xxCampaignVisible are asynchronously populated, I guess we just have to assume that it's populated by the time we as for isFeatureEnabled right?

otherwise we can make this suspending too, and have some sort of guarantee that it's waiting for the init{} function to finish first.

Reply:
Yeah we assume the network call har returned. We should probably do this behind a splash screen since its likely that we will depend on some features being fetched when the app starts. But doing this for now should be fine.